### ISSUE TITLE
Perf regression in @ngtools/webpack 8.0.2

### ISSUE THREAD

**Issue by mattlewis92** [angular/angular-cli#14719]
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [ ] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

 Yes, the previous version in which this bug was not present was: 8.0.1

### Description
When upgrading to the latest cli I noticed that our local rebuild times in JIT went from ~7s to ~24s. I managed to narrow it down to this commit: https://github.com/angular/angular-cli/commit/d61b2521923c17053d727bdd0ea05b04c93d038e Manually reverting in the source of ngtools/webpack and the re-build times went back to normal

## 🔬 Minimal Reproduction

Sorry I can't really share our repo, but I hope that it's enough to identify the commit that caused the regression.

## 🔥 Exception or Error
N/A

## 🌍 Your Environment
<pre><code>
Angular CLI: 8.0.2
Node: 12.4.0
OS: darwin x64
Angular: 8.0.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.2
@angular-devkit/build-angular     0.800.2
@angular-devkit/build-optimizer   0.800.2
@angular-devkit/build-webpack     0.800.2
@angular-devkit/core              8.0.2
@angular-devkit/schematics        8.0.2
@angular/cli                      8.0.2
@ngtools/webpack                  8.0.2
@schematics/angular               8.0.2
@schematics/update                0.800.2
rxjs                              6.5.2
typescript                        3.4.5
webpack                           4.30.0
</code></pre>

**Anything else relevant?**
N/A

**Comment by alan-agius4** [comment]
Hi, can you confirm if the rebuilds increase only when the previous compilation had an error, or the increase is constant for you?

Thanks.

**Comment by mattlewis92** [comment]
The increase is constant for me, every build previously succeeded without error.

**Comment by alan-agius4** [comment]
Thanks for the info. That is strange cause that flag should get set to false upon a successful build and true upon an invalid build.

I’ll try to look at it tomorrow and see if I am able to replicate.

**Comment by mattlewis92** [comment]
Sweet, thanks! Let me know if you can't reproduce and I'll debug locally to see why that flag's being set to true.

**Comment by mattlewis92** [comment]
Thanks for fixing this so fast! 😄

**Comment by kondi** [comment]
The fix does not look perfect, see #14775
